src/components/POLine/RoutingList/RoutingListUsers/RoutingListUsers.js

```diff
@@ -0,0 +1,135 @@
+import {
+  useCallback,
+  useMemo,
+} from 'react';
+import {
+  keyBy,
+  map,
+} from 'lodash';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  useToggle,
+  useUsersBatch,
+} from '@folio/stripes-acq-components';
+import {
+  Pluggable,
+  useStripes,
+} from '@folio/stripes/core';
+import {
+  Button,
+  Col,
+  ConfirmationModal,
+  List,
+  Loading,
+  Row,
+} from '@folio/stripes/components';
+
+import { RoutingListUserItem } from './RoutingListUserItem';
+
+import css from './RoutingListUsers.css';
+
+export const RoutingListUsers = ({
+  editable,
+  onAddUsers,
+  ids,
+}) => {
+  const stripes = useStripes();
+  const [isUnassignUsersModalVisible, toggleUnassignUsersModal] = useToggle(false);
+
+  const { isLoading, users } = useUsersBatch(ids, { keepPreviousData: true });
+
+  const onSelectUsers = useCallback((selectedUsers) => {
+    const addedUserIds = new Set(ids);
+    const newUserIds = map(selectedUsers.filter(({ id }) => !addedUserIds?.has(id)), 'id');
+
+    if (newUserIds.length) {
+      onAddUsers([...ids, ...newUserIds]);
+    }
+  }, [onAddUsers, ids]);
+
+  const onRemoveUser = useCallback((userId) => {
+    onAddUsers(ids.filter((id) => id !== userId));
+  }, [onAddUsers, ids]);
+
+  const onUnassignAllUsers = useCallback(() => {
+    onAddUsers([]);
+    toggleUnassignUsersModal();
+  }, [onAddUsers, toggleUnassignUsersModal]);
+
+  const selectedUsersMap = useMemo(() => (ids?.length ? keyBy(users, 'id') : {}), [users, ids]);
+
+  const renderItem = useCallback((user) => (
+    <RoutingListUserItem
+      canRemove={editable}
+      onRemove={onRemoveUser}
+      user={user}
+    />
+  ), [editable, onRemoveUser]);
+
+  if (isLoading) return <Loading />;
+
+  return (
+    <>
+      <Row>
+        <Col xs={12}>
+          <List
+            items={ids?.length ? users : []}
```

Comment:
`useUsersBatch` returns `users` with the `[]` as the default value, doesn't it? https://github.com/folio-org/stripes-acq-components/blob/d6175baa3baaed5c99bc15f972ca9f94152997a7/lib/hooks/useUsersBatch/useUsersBatch.js#L37

Reply:
Right. Because `{ keepPreviousData: true }` has been used as an option it will keep the latest users' list even with empty user ids.